aws/rust-runtime/aws-auth-providers/test-data/assume-role-tests.json

```diff
@@ -0,0 +1,416 @@
+[
+  {
+    "docs": "basic test case, a role_arn backed by a static credential",
+    "input": {
+      "profile": {
+        "A": {
+          "role_arn": "arn:aws:iam::123456789:role/RoleA",
+          "source_profile": "B"
+        },
+        "B": {
+          "aws_access_key_id": "abc123",
+          "aws_secret_access_key": "def456"
+        }
+      },
+      "selected_profile": "A"
+    },
+    "output": {
+      "ProfileChain": [
+        {
+          "AccessKey": {
+            "access_key_id": "abc123",
+            "secret_access_key": "def456"
+          }
+        },
+        {
+          "AssumeRole": {
+            "role_arn": "arn:aws:iam::123456789:role/RoleA"
+          }
+        }
+      ]
+    }
+  },
+  {
+    "docs": "ignore explicit credentials when source profile is specified",
+    "input": {
+      "profile": {
+        "A": {
+          "aws_access_key_id": "abc123",
+          "aws_secret_access_key": "def456",
+          "role_arn": "arn:aws:iam::123456789:role/RoleA",
+          "source_profile": "B"
+        },
+        "B": {
+          "aws_access_key_id": "ghi890",
+          "aws_secret_access_key": "jkl123"
+        }
+      },
+      "selected_profile": "A"
```

Comment:
Should we log a warning when encountering this case? Or is there a valid use-case that takes advantage of this prioritization?

Reply:
that's a good call. In general I'm planning on setting up a nice logging message that clearly talks about the chain that will be used. It's probably worth logging whenever we make a priority choice as well